store/types/validity.go

```diff
@@ -1,15 +1,30 @@
 package types
 
+import math "math"
+
+const (
+	// 32K
+	MaxKeyLength = math.MaxUint16
+	// 2G
+	MaxValueLength = math.MaxUint32
+)
+
 // AssertValidKey checks if the key is valid(key is not nil)
 func AssertValidKey(key []byte) {
 	if len(key) == 0 {
 		panic("key is nil")
 	}
+	if len(key) > MaxKeyLength {
+		panic("key is too large")
+	}
 }
 
 // AssertValidValue checks if the value is valid(value is not nil)
 func AssertValidValue(value []byte) {
 	if value == nil {
 		panic("value is nil")
 	}
+	if len(value) > MaxValueLength {
+		panic("value is too large")
+	}
```

Comment:
Constraint on value should be reflected in doc comment

Reply:
good catch, done